File: minseo/week7/지폐 접기/지폐 접기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> wallet, vector<int> bill) {
    
    int answer = 0;
   
    int wallet_width = min(wallet[0], wallet[1]);
    int wallet_height = max(wallet[0], wallet[1]);
    int bill_width = min(bill[0], bill[1]);
    int bill_height = max(bill[0], bill[1]);

    while (bill_width > wallet_width || bill_height > wallet_height) {
        
        if (bill_height > bill_width) {
            bill_height /= 2;
        } 
        else {
            bill_width /= 2;
        }
        answer++;

        
        if (bill_width > bill_height) {
            swap(bill_width, bill_height);
        }
    }

    return answer;
}

File: jinu/week6/지폐접기/지폐접기.java
class Solution {
    public int solution(int[] wallet, int[] bill) {
        int answer = 0;

        while(true){
            int wallet_min = Math.min(wallet[0],wallet[1]);
            int wallet_max = Math.max(wallet[0],wallet[1]);
            int bill_min = Math.min(bill[0],bill[1]);
            int bill_max = Math.max(bill[0],bill[1]);

            if(bill_min > wallet_min || bill_max > wallet_max){
                if(bill[0] > bill[1]){
                    bill[0] /= 2;
                }
                else
                    bill[1] /= 2;
            }
            else break;

            answer++;
        }


        return answer;
    }
}
